plugins/shadow-block-converter/src/shadow_block_converter.ts

```diff
@@ -122,17 +143,154 @@ export class BlockShadowChange extends Blockly.Events.BlockBase {
       );
     }
 
-    const value = forward ? this.newValue : this.oldValue;
-    block.setShadow(!!value);
+    const connections: Blockly.Connection[] = block.getConnections_(true);
+    if (
+      this.connectionIndex === undefined ||
+      this.connectionIndex == -1 ||
+      this.connectionIndex >= connections.length
```

Comment:
I was going to comment on the use of `==` instead of `===` but I realised we could be more defensive: 
```suggestion
      typeof this.connectionIndex !== 'number' ||
      this.connectionIndex < 0 ||
      this.connectionIndex >= connections.length
```